packages/xo-web/src/common/custom-text-input.js

```diff
@@ -0,0 +1,66 @@
+import Button from 'button'
+import decorate from 'apply-decorators'
+import Icon from 'icon'
+import PropTypes from 'prop-types'
+import React from 'react'
+import Tooltip from 'tooltip'
+import { injectState, provideState } from 'reaclette'
+import { toggleState } from 'reaclette-utils'
+
+const ENTER_KEY_CODE = 13
+const ESCAPE_KEY_CODE = 27
+
+const CustomTextInput = decorate([
+  provideState({
+    initialState: () => ({
+      editing: false,
+    }),
+    effects: {
+      closeEdition: () => ({ editing: false }),
+      onKeyDown: (effects, event) => (_, props) => {
+        const { keyCode, target } = event
+
+        if (keyCode === ENTER_KEY_CODE) {
+          if (target.value !== '') {
+            props.handler(target.value)
+            target.value = ''
+          }
```

Comment:
two nested if statements
```js
if (keyCode === ENTER_KEY_CODE && target.value !== '') {
    props.handler(target.value)
     target.value = ''
 }
```

Reply:
But I want to check the `keyCode` and ignore the` target.value` for the other conditions below (else / else if).